ruby/rspec/style_guide.md

```diff
@@ -0,0 +1,645 @@
+    # ...
+  end
+end
+```
+
+#### Good Example
+
+```ruby
+describe Article do
+  subject { create(:some_article) }
+  let(:user) { create(:user) }
+
+  before do
+    # ...
+  end
+
+  after do
+    # ...
+  end
+
+  describe '#summary' do
+    # ...
+  end
+end
+```
+
+### `it` block line returns
+
+Leave one line return around `it` blocks. This helps to separate the
+expectations from their conditional logic (contexts for instance).
+
+#### Bad Example
+
+```ruby
+describe '#summary' do
+  let(:item) { double('something') }
+
+  it 'returns the summary' do
+    # ...
+  end
+  it 'does something else' do
+    # ...
+  end
+  it 'does another thing' do
+    # ...
+  end
+end
+```
+
+#### Good Example
+
+```ruby
+describe '#summary' do
+  let(:item) { double('something') }
+
+  it 'returns the summary' do
+    # ...
+  end
+
+  it 'does something else' do
+    # ...
+  end
+
+  it 'does another thing' do
+    # ...
+  end
+end
+```
+
+### `before(:each)`or `before`
+
+There is no need to specify `(:each)` for `before`/`after` blocks, as it is
+the default functionality. There are almost zero cases to use `before(:all)`
+anymore, but if you do find one - just write it out like `before(:all)`
+
+#### Bad Example
+
+```ruby
+describe '#summary' do
+  before(:each) do
+    subject.summary = 'something'
+  end
+end
+```
+
+#### Good Example
+
+```ruby
+describe '#summary' do
+  before do
+    subject.summary = 'something'
+  end
+end
+```
+
+### 'should' it or 'should not' in `it` statements
+
+Do not write 'should' or 'should not' in the beginning of your `it` blocks.
+The descriptions represent actual functionality - not what might be happening.
+
+#### Bad Example
+
+```ruby
+it 'should return the summary' do
+  # ...
+end
+```
+
+#### Good Example
+
+```ruby
+it 'returns the summary' do
+  # ...
+end
+```
+
+### The One Expectation
+
+Use only one expectation per example. There are very few scenarios where two
+or more expectations in a single `it` block should be used. So, general rule
+of thumb is one expectation per `it` block.
+
+#### Bad Example
+
+```ruby
+describe ArticlesController do
+  #...
+
+  describe 'GET new' do
+    it 'assigns new article and renders the new article template' do
+      get :new
+      expect(assigns[:article]).to be_a(Article)
+      expect(response).to render_template :new
+    end
+  end
+
+  # ...
+end
+```
+
+#### Good Example
+
+```ruby
+describe ArticlesController do
+  #...
+
+  describe 'GET new' do
+    it 'assigns a new article' do
+      get :new
+      expect(assigns[:article]).to be_a(Article)
+    end
+
+    it 'renders the new article template' do
+      get :new
+      expect(response).to render_template :new
+    end
+  end
+end
+```
+
+### Context Cases
+
+`context` blocks should pretty much always have an opposite negative case. It
+should actually be a strong code smell if there is a single context (without a
+matching negative case) that it needs refactoring, or may have no purpose.
```

Comment:
Remove the "it" from "...it needs refactoring". The "refactoring" refers to the "single context", right?
It should be: "...a single context (...) that needs refactoring, or may have no purpose."